### PR TITLE
Ticket6079 Aeroflex Tests

### PR DESCRIPTION
**Pull request by Scott-James-Hurley**
Adds tests for the Aeroflex IOC. Similar in design to the Danfysik tests to allow for device switching. The tests cover getting and setting PVs as well as the calc records.

ISISComputingGroup/IBEX/issues/6079

### CODE REVIEW

File: common_tests/aeroflex.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        self.ca.set_pv_value('CARRIER_FREQ', 1000)
+        self.ca.assert_that_pv_is('CARRIER_FREQ', 1000)
+        
+        self.ca.set_pv_value('CARRIER_FREQ_UNITS', 'kHZ')
+        self.ca.assert_that_pv_is('CARRIER_FREQ_UNITS', 'kHZ')
+        self.ca.set_pv_value('CALC_RETURN_CAR_FREQ.PROC', 1)
+        
+        self.ca.assert_that_pv_is('CAR_FREQ_CONV', 1)
+
+    @parameterized.expand([('Value 1', 1), ('Value 2', 2), ('Value 3', 3.33333)])
```
Comment (NikolaRoev) [inline common_tests/aeroflex.py:47]: Do we need to check if the precision is handled correctly on the different devices?

**Review by NikolaRoev — CHANGES_REQUESTED** [review]
Issues suggested in my comments are in multiple tests.

These tests generally confirm that existing functionality works correctly. That's good but we need to check if the IOC will behave as we expect when things go wrong.

As an example there should be a test that sends a pulse on the 2030 in mode `AM`, will that test fail as expected?

From the ticket `There is also a 1500ms wait in the 2030 code when sending this value`, is that behaviour needed? If yes, there should be a test.

Should also probably add disconnection logic test, check out usage of `self._lewis.backdoor_simulate_disconnected_device`.

File: common_tests/aeroflex.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        self.ca.assert_that_pv_is('CARRIER_FREQ:SP_NO_ACTION', 1.2)
+        self.ca.set_pv_value('CARRIER_FREQ_UNITS:SP', 'kHZ')
+        self.ca.assert_that_pv_is('CARRIER_FREQ_UNITS:SP', 'kHZ')
+        self.ca.set_pv_value('SEND_CAR_FREQ_PARAMS.PROC', 1)
+        
+        self.ca.assert_that_pv_is('CARRIER_FREQ', 1200)
+        
+    @skip_if_recsim('Requires emulator logic so not supported in RECSIM')
+    def test_GIVEN_carrier_freq_WHEN_set_carrier_freq_readout_units_THEN_carrier_freq_readout_changes(self): 
+        self.ca.set_pv_value('CARRIER_FREQ', 1000)
```
[the remark this answers is not in the document]
Comment (Scott-James-Hurley) [reply common_tests/aeroflex.py:38]: This test checks that the carrier frequency value retrieved from the device (`CARRIER_FREQ` record) is correctly converted to the chosen units.

File: common_tests/aeroflex.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc(EMULATOR_NAME, DEVICE_PREFIX)
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX, default_wait_time=0.0)
+        
+class AeroflexCommon(AeroflexBase):
+    @skip_if_recsim('Requires emulator logic so not supported in RECSIM')
+    def test_GIVEN_new_carrier_freq_WHEN_set_carrier_freq_THEN_new_carrier_freq_set(self):        
+
+        self.ca.set_pv_value('CARRIER_FREQ:SP_NO_ACTION', 1.2)
+        self.ca.assert_that_pv_is('CARRIER_FREQ:SP_NO_ACTION', 1.2)
```
[the remark this answers is not in the document]
Comment (Scott-James-Hurley) [reply common_tests/aeroflex.py:29]: I think it's useful as this assertion makes it easier to determine where an issue lies if the test fails. I'm fine with removing it, though.

**Pull request closed without merging**